nvchecker/get_version.py

```diff
@@ -13,13 +14,45 @@
   'anitya',
 )
 
+def substitute_version(version, name, conf):
+  '''
+  Substitute the version string via defined rules in the configuration file.
+  Accepted rules are:
+
+  * from_pattern, to_pattern: Both should be Python regular expressions.
+    `from_pattern` found in the version string will be substituted to
+    `to_pattern`.
+  * prefix: If the version string starts with `prefix`, the prefix is removed.
+    Otherwise the version string is returned as-is.
+
+  If both prefix and from_pattern/to_pattern are used, from_pattern/to_pattern
+  are ignored.
+  '''
+  prefix = conf.get('prefix')
+  if prefix:
+    if version.startswith(prefix):
+      version = version[len(prefix):]
+    return version
+
+  from_pattern = conf.get('from_pattern')
+  if from_pattern:
+    to_pattern = conf.get('to_pattern')
+    if not to_pattern:
+      raise ValueError('%s: from_pattern exists but to_pattern doesn\'t', name)
+
+    return re.sub(from_pattern, to_pattern, version)
+
+  # No substitution rules found. Just return the original version string.
+  return version
+
 async def get_version(name, conf):
   for key in handler_precedence:
     if key in conf:
       func = import_module('.source.' + key, __package__).get_version
       version = await func(name, conf)
       if version:
-        version.replace('\n', ' ')
+        version = version.replace('\n', ' ')
+        version = substitute_version(version, name, conf)
```

Comment:
Need to catch exceptions (ValueError and sre_constants.error) and log them here.